lms/extensions/feature_flags/_feature_flags.py

```diff
@@ -39,12 +39,13 @@ def flag_is_active(self, request, feature_flag_name):
         otherwise. All feature flags are ``False`` by default (if no source
         toggles the feature flag either on or off, it will be off by default).
         """
-        results = [False]  # All feature flags are False by default.
-        results.extend(
-            [provider(request, feature_flag_name) for provider in self._providers]
-        )
-        results = [result for result in results if result is not None]
-        return results[-1]
+
+        for provider in reversed(self._providers):
+            enabled = provider(request, feature_flag_name)
+            if enabled is not None:
+                return enabled
+
+        return False
```

Comment:
This does exactly the same thing except it doesn't bother providers lower down the chain if one further on gives a definitive answer. Previously we asked them all and discarded the answers. This necessitated a test change.

Reply:
I changed this was mostly because it confused me rather than efficiency, but it's a nice bonus!